src/pipeline-resolver.cpp

```
@@ -1,50 +1,417 @@
+                {
+                    auto* payload = new Payload{m_device, program.program, &entryPoint};
+                    batch.submit(
+                        [](void* data)
+                        {
+                            auto* payload = static_cast<Payload*>(data);
+                            // TODO: Slang exposes only cumulative global compiler timing. Parallel
+                            // entry-point compilation therefore cannot currently attribute compiler
+                            // time accurately to individual compilation reports.
+                            payload->entryPoint->result =
+                                payload->program->compileEntryPoint(payload->device, *payload->entryPoint, false);
+                        },
```

Cppcheck (2.21.0)

[portability] Shifting signed 32-bit value by 31 bits is implementation-defined behaviour

(shiftTooManyBitsSigned)